.github/workflows/tests.yml

```diff
@@ -20,7 +20,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest]
-        python-version: [3.7, 3.8, 3.9]
+        python-version: [3.7, 3.8, 3.9, '3.10']
```

Comment:
@slarse I did not add 3.10 to the `exclude` list below just to see what happens. It seems that mac-OS job keeps on running without termination. I will remove the OS from the matrix itself, in that case. Do we also need to remove the `macOS` badge from the README?

Reply:
There's a mistake here from #988, we want to run macOS tests for the _earliest_ supported version (3.7), as if it runs on 3.7 it generally runs on anything after that (and we have the Ubuntu tests for all supported versions). So could you add 3.10 to the exclude list and remove 3.7 from said list?

> Why aren't we running macOS tests for other Python versions?

Because the macOS tests are super slow, and we're really just testing for platform compatibility. It's incredibly unlikely that we'd have all tests green on Ubuntu, and on macOS with 3.7, but _not_ on macOS with say.. 3.8. It could theoretically happen, but the chances are so remote that I don't think it's worth actively monitoring for.